doc/developers/contributing.rst

```diff
@@ -796,11 +796,28 @@ In the following example, k is deprecated and renamed to n_clusters::
                           "will be removed in 0.15.", DeprecationWarning)
             n_clusters = k
 
+If the default value of a parameter needs to be changed, it is recommended to
+replace the default value with a specific value (e.g., ``deprecated``) and
+raise ``FutureWarning`` when users are using the default value. In the
+following example, we change the default value of ``n_clusters`` from 5 to 10
+(current version is 0.20). You can also refer to recent merged PRs
+(e.g., `#10331 <https://github.com/scikit-learn/scikit-learn/pull/10331>`__
+and `#11043 <https://github.com/scikit-learn/scikit-learn/pull/11043>`__)::
+
+    import warnings
+
+    def example_function(n_clusters='deprecated'):
+        if n_clusters == 'deprecated':
+            warnings.warn("The default value of n_clusters will change from "
+                          "5 to 10 in 0.22.", FutureWarning)
+            n_clusters = 5
+
```

Comment:
I think there should also be an example for an ``__init__`` parameter in a class to show that the deprecation needs to happen in ``fit..